### DISCUSSION TITLE
Step forward in increments other than 100ms

### DISCUSSION THREAD

**Discussion by vivek-rane** [foxglove/community#672] [Q&A]
The current step forward seems to be tailored to LiDAR based systems and moves 100ms at a time. We would like to set the rate to a different frequency (30Hz) so we can debug one step at a time without constantly having to click the mouse and seek.

This might be a pretty simple change - please help!

**Comment by jtbandes** [comment]
Holding down the shift key while seeking will seek by 10ms:

https://github.com/foxglove/studio/blob/f674bbfb67c72396c730b072a60b3685b914e317/packages/studio-base/src/components/PlaybackControls/sharedHelpers.ts#L18

**Reply by vivek-rane** [reply]
I'm really looking for 33ms (for 30Hz) but that is very helpful in the meantime!

**Comment by jameskuszmaul-brt** [comment]
Copying a conversation on this from the community slack for posterity:
![image](https://user-images.githubusercontent.com/91565539/234367050-53020130-9a99-45ce-85be-98a34f65f947.png)

Copying my comments for better consumability:

> I could envision use-cases for a combination of these. I think that most use-cases I've had it would be appropriate & principled to choose a particular topic and use that for my stepping. However, sometimes there are situations where either (a) I am too lazy to figure out what that topic is and/or (b) I need to debug an issue with the system where I want to observe every single thing that happens on any topic that I'm visualizing, and I want that to include every single dependency of the visualizations that I'm using. In those situations, having the option to do a "next new information of any time" would be helpful.
> E.g., for "tie to one signal" I may be plotting 10+ different messages that are all happening at the same period and just want to step forward by one period at once, and so just choose a single signal.
On the other hand, if you e.g. had a complex scene for your autonomous vehicle with many different signals and you are trying to figure out exactly what changed to cause your vehicle to decide to start braking, you may want to step through every single message you can find around the relevant timepoint.
